Modules/Registration/Common/include/itkLandmarkBasedTransformInitializer.hxx

```diff
@@ -355,71 +361,180 @@ LandmarkBasedTransformInitializer<TTransform, TFixedImage, TMovingImage>::Intern
     itkExceptionMacro("Transform is VersorRigid3DTransform and Moving image dimension is not 3");
   }
 
-  // --- compute the necessary transform to match the two sets of landmarks
-  // ---
-  //
-  //
-  //    The solution is based on
-  //    Berthold K. P. Horn (1987),
-  //    "Closed-form solution of absolute orientation using unit
-  // quaternions,"
-  //    Journal of the Optical Society of America A, 4:629-642
-  //
-  //
-  //    Original python implementation by David G. Gobbi
-  //    Readpted from the code in VTK: Hybrid/vtkLandmarkTransform
-  //
-  //----------------------------------------------------------------------------
+  /*----------------------------------------------------------------------------
+     Compute the necessary transform to match the two sets of landmarks
+
+     The solution is based on
+     Berthold K. P. Horn (1987),
+     "Closed-form solution of absolute orientation using unit quaternions,"
+     Journal of the Optical Society of America A, 4:629-642
+
+     Original python implementation by David G. Gobbi
+     Readapted from the code in VTK: Hybrid/vtkLandmarkTransform
+  ----------------------------------------------------------------------------*/
 
   using VectorType = typename VersorRigid3DTransformType::OutputVectorType;
-  using PointType = typename VersorRigid3DTransformType::OutputPointType;
 
   // Compute the centroids
-  PointType fixedCentroid;
-  fixedCentroid.Fill(0.0);
-  PointsContainerConstIterator fixedItr = m_FixedLandmarks.begin();
-  while (fixedItr != m_FixedLandmarks.end())
+  PointType3D fixedCentroid = ComputeCentroid(m_FixedLandmarks);
+
+  PointType3D movingCentroid = ComputeCentroid(m_MovingLandmarks);
```

Comment:
The ITK SW Guide recommends using `this->` to access member methods (p. 333) (applies to any code in the PR; have not checked all calls).